### ISSUE TITLE
Different size of pages

### ISSUE THREAD

**Issue by Zanexess** [barteksc/AndroidPdfViewer#213]
I have the pdf like this
```
          ---------
          |       |
          |       |  -----------
          |       |  |          |
-------   |       |  |          |
|      |  |       |  |          |
|      |  |       |  |          |
-------   |       |  |          |
          |       |  |          |
          |       |  -----------
          |       |
          |       |
          ---------
```

And so on.
But i want to display it like according to width of my phone to be able to move among vertical axis on current page.
```
-------  --------  --------
|       ||       ||       |
|       ||       ||       |
------- ||       ||       |
        ||       ||       |
        ||       |--------
        ||       |
          --------

Is there any solutions?

**Comment by 1stmetro** [comment]
I guess the answer is to resize the pdfview display on every page change, maybe too intensive so a button of some description may be the answer, long press event etc..

**Comment by Zanexess** [comment]
I tried to calculate optimal page width and height in onPageChanged() method, but redrawing all page in OnPageListener gives bad user experience, moreover it doesn't solve anything.
The problem here that in that library we assume that all pages have the equal height and width.
Just try to use pdf where all pages have equal width, but diffenent height. For example first page is 1000*1000, but second 1000 * 8000px and so on;
I also tried to modify the code like #97. It's good solution if we using vertical mode, but there is an issue when I zoom page. The page renders only in top and gives simmilar effect like in this video https://www.youtube.com/watch?v=QApCYasyzRA. 
Changing the Constants and the width and height of the row and column in the code (see #135) cause the OutOfMemory exception on some devices.
So I don't understand how to avoid it and achieve good result of rendering when pdf have different height of pages. I think that the library with 1.7k stars should provide this functionality. Do you have any ideas?

**Comment by 1stmetro** [comment]
See my post from earlier today with regards to into sizing pages, should address this issue too.

**Comment by vadimkorovinsky** [comment]
Hi ! Does anyone solved this problem ? When page sizes are different, document looks bad.

**Comment by 1stmetro** [comment]
Use the code I posted it will be in the closed section

**Comment by Zanexess** [comment]
@1stmetro 
doesn't work when height is bigger than width
![photo_2017-04-03_14-28-41](https://cloud.githubusercontent.com/assets/9991309/24607387/123ce0a0-187a-11e7-8702-abd5b6e3458a.jpg)
![photo_2017-04-03_14-28-50](https://cloud.githubusercontent.com/assets/9991309/24607388/123cd416-187a-11e7-87f3-db7573e40e8d.jpg)

**Comment by 1stmetro** [comment]
Already fixed this check my fork also added page markers in custom width and Colours.

**Comment by 1stmetro** [comment]
You can tell my code to auto size on height or width by setting the orientation constant to 1 or 0 depending on how you want it sized so just a case of knowing device orientation or page orientation and setting the blue as I indicated.

**Comment by Zanexess** [comment]
I have tried it already, but the problem is still here

![photo_2017-04-03_15-39-43](https://cloud.githubusercontent.com/assets/9991309/24609532/cf0cf2c0-1883-11e7-9e2d-61cacdee40cb.jpg)
![photo_2017-04-03_15-39-49](https://cloud.githubusercontent.com/assets/9991309/24609533/cf263762-1883-11e7-8204-e10003b49ed1.jpg)

**Comment by 1stmetro** [comment]
You need to reverse the code and add 2 more constants  or choose which way to size depending if width bigger than height etc.  Not near a pc currently to amend, care to post your pdf example?

**Comment by vadimkorovinsky** [comment]
@1stmetro , this is build from your fork, pages have disproportion
![pdf1](https://cloud.githubusercontent.com/assets/25370740/24610129/7b7c9e5c-188e-11e7-84cf-f763ed4ca135.png)

**Comment by Zanexess** [comment]
@1stmetro 
This pdf is not for distribution, that's why I've invited you to private repository with the file.

**Comment by 1stmetro** [comment]
The car pdf using an orientation of 1 or 0? Remember one does by width one by height... you pass the orientation for which way to auto size it doesn't pick which way to auto size default is 0 so set the constant to change the width or height sizing.

**Comment by 1stmetro** [comment]
Once i load your example pdf will simple enough to sort the code once I see what's going on.

**Comment by vadimkorovinsky** [comment]
Yes, I can post sample pdf now, but can not find any contacts here.

**Comment by 1stmetro** [comment]
just upload anywhere i will grab a copy..

**Comment by 1stmetro** [comment]
After some time and playing around, here is the answer.. with a question.

The code below demonstrates the use of setScaleX and Y to get it readable, but i have an issue with page lengths if you enable page borders you will notice all the pages have the same height so how do you translate that to get the setScale ratios...... presume you can get the page height by rewriting the page height code and translate into ratios but not had time to look further...

try x=1 or 2 and y=3 or 6 and scroll through pages

if (pageOrientation == 0) {
                int centerPos = getPageAtPositionOffset(0);
               // zoomTo(getWidth() / optimalPageWidth);
                setScaleX(2);
                setScaleY(6);
                setPositionOffset(centerPos);
            }

**Comment by vadimkorovinsky** [comment]
Car pdf: https://cloud.mail.ru/public/B4tT/Jj4118NzJ

**Comment by maxbax** [comment]
I have the same problem: a pdf with some pages with different sizes. The library show all pages with the size of the first page.

**Comment by bharper1** [comment]
The library is absolutely fantastic. Are there any plans to update it so that true page sizes are displayed correctly with the right orientation?
